src/main/scala/com/softwaremill/realworld/articles/ArticlesService.scala

```diff
@@ -1,16 +1,16 @@
 package com.softwaremill.realworld.articles
 
 import com.softwaremill.realworld.profiles.ProfileRow
-import com.softwaremill.realworld.utils.Exceptions
+import com.softwaremill.realworld.utils.{Exceptions, Pagination}
 import zio.{IO, ZIO, ZLayer}
 
 import java.sql.SQLException
 import javax.sql.DataSource
 
 class ArticlesService(articlesRepository: ArticlesRepository):
 
-  def list(): ZIO[Any, Nothing, List[Article]] = articlesRepository
-    .list()
+  def list(filters: Map[ArticlesFilters, String], pagination: Pagination): ZIO[Any, Nothing, List[Article]] = articlesRepository
```

Comment:
The type can be made shorter by using `UIO[List[Article]]`. Though it's suspicious that a DB operation can't throw exceptions

Reply:
They throw `SQLException`, but I treat those as defects (`.orDie` in repo), that are logged and returned as 500, as we can't do anything with them.

Reply:
Hm but it's not a defect - that is, not a bug that can be fixed. So I think in the ZIO philosophy this would be in the error channel?

Reply:
Ok, I understood it differently. I was looking at https://tapir.softwaremill.com/en/latest/server/errors.html and `SQLException` seemed to fit to `2. Failed effects: exceptions which are not handled by the server logic (corresponds to 5xx responses` and I thought that every 500 could be interpreted as zio defect.

Reply:
Yeah ZIO is a bit different here as they have the dedicated error channel. A "failed effect" in case of ZIO is any case when the resulting effect contains an error (not necessarily a defect, might be in the error channel). So this would still translate to a 500.